example-utils/integration-test-utils/src/integration_test_utils.rs

```diff
@@ -0,0 +1,335 @@
+                let curr_msb = msg.attributes.as_ref().unwrap().id.msb;
+                debug!("prev_msb: {prev_msb}, curr_msb: {curr_msb}");
+                if curr_msb <= prev_msb {
+                    panic!("!! -- Message ordering issue for lsb: {} -- !!", lsb);
+                }
+                prev_msb = curr_msb;
+            }
+        }
+    }
+}
+
+#[inline(always)]
+fn override_lsb_rand_b(lsb: u64, new_rand_b: u64) -> u64 {
+    lsb & CLEAR_RAND_B | new_rand_b
+}
+
+pub async fn wait_for_pause(signal: Signal) {
+    let (lock, cvar) = &*signal;
+    let mut has_paused = lock.lock().await;
+    while !*has_paused {
+        debug!("inside wait_for_pause entered while loop");
+        // Wait until the client signals it has paused
+        has_paused = cvar.wait(has_paused).await;
+        debug!("received has_paused notification");
+    }
+    debug!("exiting wait_for_pause");
+}
+
+pub async fn signal_to_pause(signal: Signal) {
+    let (lock, cvar) = &*signal;
+    let mut should_pause = lock.lock().await;
+    *should_pause = true; // Indicate the client should pause
+    cvar.notify_all(); // Wake up the client so it can check the condition and continue
+}
+
+pub async fn signal_to_resume(signal: Signal) {
+    let (lock, cvar) = &*signal;
+    let mut should_pause = lock.lock().await;
+    *should_pause = false; // Indicate the client should no longer pause
+    cvar.notify_all(); // Wake up the client so it can check the condition and continue
+}
+
+pub async fn reset_pause(signal: Signal) {
+    {
+        let (lock, _cvar) = &*signal;
+        let mut has_paused = lock.lock().await;
+        *has_paused = false;
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn run_client(
+    name: String,
+    my_client_uuri: UUri,
+    _other_client_uuri: UUri,
+    listener: Arc<dyn UListener>,
+    tx: Sender<Result<UMessage, UStatus>>,
+    rx: Receiver<Result<UMessage, UStatus>>,
+    _publish_msgs: Vec<UMessage>,
+    mut notification_msgs: Vec<UMessage>,
+    mut request_msgs: Vec<UMessage>,
+    mut response_msgs: Vec<UMessage>,
+    send: bool,
+    pause_execution: Signal,
+    execution_paused: Signal,
+    client_command: Arc<Mutex<ClientCommand>>,
+    number_of_sends: Arc<AtomicU64>,
+    sent_message_vec_capacity: usize,
+) -> JoinHandle<Vec<UMessage>> {
+    std::thread::spawn(move || {
+        task::block_on(async move {
+            let client = UPClientFoo::new(&name, rx, tx).await;
+
+            let client_rand_b = random::<u64>() >> 2;
+
+            let mut sent_messages = Vec::with_capacity(sent_message_vec_capacity);
+
+            let register_res = client
+                .register_listener(my_client_uuri.clone(), listener)
+                .await;
+            let Ok(_registration_string) = register_res else {
+                panic!("Unable to register!");
+            };
+
+            let mut active_connection_listing = Vec::new();
+
+            // let register_res = client
```

Comment:
For same reasoning as [here](https://github.com/eclipse-uprotocol/up-streamer-rust/pull/8/files/a53b62db291a2e74dd6b5022863b802db36a2bc5#r1582008874) this should also be removed.

Reply:
Done